cmd/attest_test.go

```diff
@@ -0,0 +1,199 @@
+package cmd
+
+import (
+	"os"
+	"testing"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/internal/test"
+)
+
+func makeOutputFile(tb testing.TB, output string) string {
+	tb.Helper()
+	file, err := os.CreateTemp("", output)
+	if err != nil {
+		tb.Fatal(err)
+	}
+	defer file.Close()
+	return file.Name()
+}
+
+func TestNonce(t * testing.T){
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	//without nonce
+	RootCmd.SetArgs([]string{"attest","--key","AK"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
+	}
+
+	//with odd length nonce
+	RootCmd.SetArgs([]string{"attest","--nonce","12345","--key","AK"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
+	}
+
+}
+
+func TestAttestPass(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	tests := []struct {
+		name          string
+		key           string
+		algo          string
+		nonce         string
+		teenonce      string
+		outputText    string
+	}{
+		{"DefaultKey","","rsa","2222","",""},
+		{"Withnonce", "AK", "rsa", "2222", "", ""},
+		{"WithTextOutput", "AK", "rsa", "2222", "", "text1111"},
+		//@TODO These cases works when we are able to create gce key with ExternalTPM.
+		// {"WithNonce","gceAK","rsa","2222","",""},
+		// {"WithTextOutput","gceAK","rsa","2222","","text1111"},
+	}
+
+	for _, op := range tests {
+		t.Run(op.name, func(t *testing.T) {
+			secretFile1 := makeOutputFile(t, "attest.binarypb")
+
+			attestArgs := []string{"attest", "--output", secretFile1}
+			if op.key != "" {
+				attestArgs = append(attestArgs, "--key", op.key)
+			}
+			if op.algo != "" {
+				attestArgs = append(attestArgs, "--algo", op.algo)
+			}
+			if op.nonce != "" {
+				attestArgs = append(attestArgs, "--nonce", op.nonce)
+			}
+			if op.teenonce != "" {
+				attestArgs = append(attestArgs, "--teenonce", op.teenonce)
+			}
+			if op.outputText != "" {
+				attestArgs = append(attestArgs, "--outputText", op.outputText)
+			}
+
+			RootCmd.SetArgs(attestArgs)
+			if err := RootCmd.Execute(); err != nil {
+				t.Error(err)
+			}
+
+		})
+	}
+}
+
+func TestGceFlagFail(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+	//@TODO This case works when we are able to create gce key with ExternalTPM.
```

Comment:
Best to remove TODOs in a PR or open an issue to reference if you really can't get to it in the PR.

Reply:
Agreed , this PR is mostly to get the first review done with Alex. After concluding the discussion to clarify the functionality , I'll remove those TODOs and update you on this thread .

Reply:
I have removed all those @TODOs. 
I'll be resolving this comment.